examples/apps/generic_client/eth_generic_client.py

```diff
@@ -0,0 +1,657 @@
+#! /usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import json
+import argparse
+import logging
+import secrets
+import asyncio
+
+import config.config as pconfig
+import utility.logger as plogger
+import crypto_utils.crypto_utility as crypto_utility
+from avalon_sdk.worker.worker_details import WorkerType
+import avalon_sdk.worker.worker_details as worker_details
+from avalon_sdk.work_order.work_order_params import WorkOrderParams
+from avalon_sdk.ethereum.ethereum_worker_registry_list import \
+    EthereumWorkerRegistryListImpl
+from avalon_sdk.direct.jrpc.jrpc_worker_registry import \
+    JRPCWorkerRegistryImpl
+from avalon_sdk.direct.jrpc.jrpc_work_order import \
+    JRPCWorkOrderImpl
+from avalon_sdk.direct.jrpc.jrpc_work_order_receipt \
+    import JRPCWorkOrderReceiptImpl
+from error_code.error_status import WorkOrderStatus, ReceiptCreateStatus
+import crypto_utils.signature as signature
+from error_code.error_status import SignatureStatus
+from avalon_sdk.work_order_receipt.work_order_receipt \
+    import WorkOrderReceiptRequest
+# from avalon_sdk.fabric.fabric_worker_registry import FabricWorkerRegistryImpl
+# from avalon_sdk.fabric.fabric_work_order import FabricWorkOrderImpl
+from avalon_sdk.ethereum.ethereum_worker_registry \
+    import EthereumWorkerRegistryImpl
+from avalon_sdk.ethereum.ethereum_work_order import EthereumWorkOrderProxyImpl
+
+
+# Remove duplicate loggers
+for handler in logging.root.handlers[:]:
+    logging.root.removeHandler(handler)
+logger = logging.getLogger(__name__)
+TCFHOME = os.environ.get("TCF_HOME", "../../")
+WAIT_TIME = 31536000
+verification_key = None
+s_key = None
+s_iv = None
+wo_id = ''
+
+
+def _parse_command_line(args):
+
+    parser = argparse.ArgumentParser()
+    mutually_excl_group = parser.add_mutually_exclusive_group(required=True)
+    parser.add_argument(
+        "-c", "--config",
+        help="The config file containing the Ethereum contract information",
+        type=str)
+    mutually_excl_group.add_argument(
+        "-u", "--uri",
+        help="Direct API listener endpoint, default is http://localhost:1947",
+        type=str,
+        default="http://avalon-listener:1947")
+    mutually_excl_group.add_argument(
+        "-b", "--blockchain",
+        help="Blockchain type to use in proxy model",
+        choices={"fabric", "ethereum"},
+        type=str,
+        default="ethereum"
```

Comment:
In the future it would be good to combine the generic clients (Fabric and Ethereum) with `generic_client.py`

Reply:
Yes exactly Dan. That was the goal we started with. Then we had to do it this way because there are differences between the direct & proxy model which we scope to resolve and unify thereafter.